src/coreclr/jit/lsraxarch.cpp

```diff
@@ -3057,6 +3263,11 @@ int LinearScan::BuildMul(GenTree* tree)
         dstCount      = 2;
     }
 #endif
+    else
+    {
+        // ToDo-APX : imul currently doesn't have rex2 support. So, cannot  use R16-R31.
+        dstCandidates = BuildApxIncompatibleGPRMask(tree, true);
```

Comment:
Calls to `BuildApxIncompatibleGPRMask` for many nodes seems expensive. Wondering if we can do something like:
1. at the top just set `SingleTypeRegSet incompatibleGprMask = compiler->canUseApxEncoding() ? lowGPRRegs() : RBM_NONE;`
2. Places where you are passing `forceLowGpr= true` can instead just use `incompatibleGprMask`.
3. Places where you are not forcing lowGPr, can just use `DoesThisUseGPR(tree) ? incompatibleGprMask : RBM_NONE`

Also, might worth caching the value of `lowGPRRegs()`  because currently it is evaluated every time to be `(availableIntRegs & RBM_LOWINT.GetIntRegSet())` and I see `lowGprRegs()` is used at lot of places.